packages/frontend/src/helpers/toolbox.ts

```diff
@@ -114,3 +118,34 @@ export function areAllIfThenBranchesCompleted(
   const branches = extractBranchesWithSteps(allBranches, depth)
   return branches.every(isIfThenBranchCompleted)
 }
+
+/**
+ * Helper hook to check if If-Then action should be selectable; supports edge
+ * case in ChooseAppAndEventSubstep.
+ *
+ * If-Then should only be selectable if:
+ * - We're the last step.
+ * - We are not inside a branch (unless we're whitelisted for nested
+ *   branches via LD).
+ *
+ * Using many consts as purpose of the conditions may not be immediately
+ * apparent.
+ */
+export function useIsIfThenSelectable(isLastStep: boolean): boolean {
```

Comment:
can use namedVariable here? { isLastStep }, cos it's not intuitive as a param here

Reply:
yep agreed